src/api/multipart/multipartUpload.ts

```diff
@@ -0,0 +1,97 @@
+import {Thenable} from '../../thenable/Thenable'
+import multipartUploadPart from './multipartUploadPart'
+import {getFileSize} from './getFileSize'
+import {getChunks} from './getChunks'
+import defaultSettings from '../../defaultSettings'
+
+/* Types */
+import {FileData, SettingsInterface} from '../../types'
+import {ChunkType, MultipartPart} from './types'
+import {BaseThenableInterface} from '../../thenable/types'
+
+function throttle(callback, limit = 1) {
+  let wait = false                  // Initially, we're not waiting
+
+  return function (...args) {               // We return a throttled function
+    if (!wait) {                   // If we're not waiting
+      callback(...args)           // Execute users function
+      wait = true               // Prevent future invocations
+      setTimeout(function () {   // After a period of time
+        wait = false          // And allow future invocations
+      }, limit);
+    }
+  }
+}
+
+class MultipartUpload extends Thenable<any> implements BaseThenableInterface<any> {
```

Comment:
a lot of `any` :(

Reply:
Yes, I agree. I need to come up with something with this